### PR TITLE
bpo-30303: IDLE: Add _utest argument to textview

### PR DESCRIPTION
**Pull request by louisom**

### CODE REVIEW

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: Use global 'root' here and below.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: I don't think this and tearDownClass are needed.  The module root should work.  Just pass '_utest=True' in both calls.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: See comment below about findfile.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: After I adapt my coverage.bat to work with the git repository, I will get  new value.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: Follow with 'button.destroy()'.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: osp = os.path
fn =osp.abspath(osp.join(osp.dirname('__file__'), '..', 'CREDITS.txt'))
would be equivalent.  But to make the test simpler, just pass '__file__' instead of fn and adjust assert.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: Using  __file__ makes this file self-contained.
self.assertIn(self.view.textView.get('1.0', '1.end'), 'idlelib.textview'
If this file is edited to make this fail, then this file fails immediately.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
At first reading, the textview.py changes are just right.  The test needs some minor changes.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [reply Lib/idlelib/idle_test/test_textview.py]: The boldface __file__is the dunder name interpreted as markup.  Unlike StackOverflow, there is no preview of the rendering. Let's try it with backticks `__file__`.

Comment (terryjreedy) [reply Lib/idlelib/idle_test/test_textview.py]: Maybe the interpretation is only in quotes?  '__file__', "__file__".

Comment (louisom) [reply Lib/idlelib/idle_test/test_textview.py]: I can't reproduce the equivalent between osp and findfile. On my situation, I'm running unittest at python/cpython, so the `osp.dirname('file')` will return ``, not `Lib/idlelib/idle_test`.

Comment (terryjreedy) [reply Lib/idlelib/idle_test/test_textview.py]: Now with quotes and backticks: backtick inner: "`__file__`", outer `'__file__'`.
PS I checked the markdown link below but it was insufficient.

Comment (louisom) [reply Lib/idlelib/idle_test/test_textview.py]: ah, I see, is `__file__`, not '**file**'

File: Lib/idlelib/idle_test/test_textview.py
Comment (louisom) [reply Lib/idlelib/idle_test/test_textview.py]: I'm using `addCleanup` instead of directly using `button.destroy`

Comment (terryjreedy) [reply Lib/idlelib/idle_test/test_textview.py]: Good idea.

File: Lib/idlelib/idle_test/test_textview.py
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_textview.py]: I like making this future proof.  However, since the 2nd line is blank and Texts have a default extra \n at the end, I believe this would pass even if the textview only displayed one line.  So I added an intermediate `f.readline()` and changed 2s to 3s.

**Review by terryjreedy — APPROVED** [review]
I am about to push my changes back and will merge once CI allows.

**Comment by brettcannon** [comment]
Can I ask if a CLA was eventually verified as being signed before merging this?

**Comment by Mariatta** [comment]
@brettcannon I asked the PR author in [bpo-30303](http://bugs.python.org/issue30303#msg295234) about this.
It seems that @louisom is the same person as @mlouielu, and @mlouielu has signed the CLA.

**Comment by mlouielu** [comment]
@louisom is my previous account, does b.p.o. support comma split (louisom, mlouielu) at the detail page?

**Comment by mlouielu** [comment]
@Mariatta I've add another CLA to @louisom, please help to verify this, thanks!

**Comment by brettcannon** [comment]
@Mariatta @mlouielu thanks for the clarification!

**Comment by terryjreedy** [comment]
The backport error is because this was backported already in #1916.  (And the file has been refactored since.)  I removed the label.

**Pull request merged**
